pandas/tests/indexes/test_base.py

```diff
@@ -2613,3 +2613,28 @@ def construct(dtype):
         no_matches = np.array([-1] * 6, dtype=np.intp)
         tm.assert_numpy_array_equal(result[0], no_matches)
         tm.assert_numpy_array_equal(result[1], no_matches)
+
+
+def test_union_with_duplicate_index():
+    # GH: 36289
+    idx1 = Index([1, 0, 0])
```

Comment:
can you test this with multiple dtypes/index subclasses

Reply:
Did you have something like that in mind?